### DISCUSSION TITLE
Representing Blocks.txt?

### DISCUSSION THREAD

**Discussion by faassen** [unicode-org/icu4x#4798] [Ideas]
I dug around the source code, but I couldn't find a representation in Rust code of the `Blocks.txt` data. There's the `unicode_blocks` crate, but this misses an important feature; I need to be able to iterate through all blocks. I need to be able to look them up by name, but not the name as given as this has space characters. So iteration seems required so I can do some pre-processing. This is to implement regular expressions as defined by appendix F in the XML Schema specification:

Being able to get a `CodePointInvList` for a block would also be nice.

https://www.w3.org/TR/xmlschema-2/#regexs

Did I miss something? Is this planned?

**Comment by sffc** [comment]
Good suggestion! I opened #4803. This would be a good first issue if someone wanted to contribute this.
